README.md

```diff
@@ -18,6 +18,123 @@ to facilitate unit-testing of OpenTracing Java instrumentation.
+
+```
+io.opentracing.Tracer tracer = ...;
+...
+Span span = tracer.buildSpan("someWork").startManual();
+try {
+    // (do things / record data to `span`)
+} finally {
+    span.finish();
+}
+```
+
+Or, to take advantage of `ActiveSpanSource` and automatic intra-process propagation, like this:
+
+```
+io.opentracing.Tracer tracer = ...;
+...
+try (ActiveSpan activeSpan = tracer.buildSpan("someWork").startActive()) {
+    // Do things.
+    //
+    // If we create async work, `activeSpan.capture()` allows us to pass the `ActiveSpan` along as well.
+}
+```
+
+
+**If there is an `ActiveSpan`, it will act as the parent to any newly `startActive()`ed `Span`** unless the programmer invokes `asRoot()` at `buildSpan()` time, like so:
+
+```
+io.opentracing.Tracer tracer = ...;
+...
+SpanContext someOtherSpanContext = ...;
+ActiveSpan span = tracer.buildSpan("someWork").asRoot().startActive();
+```
+
+### Deferring asynchronous work
+
+Consider the case where a `Span`'s lifetime logically starts in one execution context and ends in another. For instance, the intra-Span timing breakdown might look like this:
+
+```
+[ ServiceHandlerSpan                                 ]
+|-FunctionA-|-----waiting on an RPC------|-FunctionB-|
+            
+------------------------------------------------> time
+```
+
+The `"ServiceHandlerSpan"` is _active_ when it's running FunctionA and FunctionB, and inactive while it's waiting on an RPC (presumably modelled as its own Span, though that's not the concern here).
+
+**The `ActiveSpanSource` makes it easy to "adopt" the Span and execution context in `FunctionA` and re-activate it in `FunctionB`.** Note that every `Tracer` must also implement `ActiveSpanSource`. These are the steps:
+
+1. Start the `ActiveSpan` via `Tracer.startActive()` rather than via `Tracer.startManual()`; or, if the `Span` was already `startManual()`ed, call `ActiveSpanSource#adopt(span)`. Either route will yield an `ActiveSpan` instance that's "adopted" the `Span`.
+2. In the method/function that *allocates* the closure/`Runnable`/`Future`/etc, call `ActiveSpan#capture()` to obtain an `ActiveSpan.Continuation`
+3. In the closure/`Runnable`/`Future`/etc itself, invoke `ActiveSpan.Continuation#activate` to re-activate the `ActiveSpan`, then `deactivate()` it when the Span is no longer active (or use try-with-resources for less typing).
+
+For example:
+
+```
+io.opentracing.Tracer tracer = ...;
+...
+// STEP 1 ABOVE: start the ActiveSpan
+try (ActiveSpan serviceSpan = tracer.buildSpan("ServiceHandlerSpan").startActive()) {
+    ...
+
+    // STEP 2 ABOVE: capture the ActiveSpan
+    final ActiveSpan.Continuation cont = serviceSpan.capture();
+    doAsyncWork(new Runnable() {
+        @Override
+        public void run() {
+
+            // STEP 3 ABOVE: use the Continuation to reactivate the Span in the callback.
+            try (ActiveSpan activeSpan = cont.activate()) {
+                ...
```

Comment:
Maybe a stupid question because I don't know much about Java. But in this example, if I have access to the outer `cont` variable in the `Runnable` (it's not passed as a parameter), why couldn't I just access the `serviceSpan` directly instead and write something like this - completely without the need of active spans, continuations etc.:

```java
someMethod() {
    final Span serviceSpan = tracer.buildSpan("ServiceHandlerSpan").start();
    
    serviceSpan.setTag("key1", "value");

    doAsyncWork(new Runnable() {
        @Override
        public void run() {
            serviceSpan.setTag("key3", "value");
            serviceSpan.finish();
        }
    }

    serviceSpan.setTag("key2", "value");
}
```
Would this not be possible?